ledger/src/blockstore_db.rs

```diff
@@ -99,6 +111,222 @@ const PROGRAM_COSTS_CF: &str = "program_costs";
+    // RocksDB's internal property key: "rocksdb.block-cache-usage"
+    pub block_cache_usage: i64,
+    // The memory size used by the column family in the block cache where
+    // entries are pinned.
+    // RocksDB's internal property key: "rocksdb.block-cache-pinned-usage"
+    pub block_cache_pinned_usage: i64,
+
+    // The estimated memory size used for reading SST tables in this column
+    // family such as filters and index blocks. Note that this number does not
+    // include the memory used in block cache.
+    // RocksDB's internal property key: "rocksdb.estimate-table-readers-mem"
+    pub estimate_table_readers_mem: i64,
+
+    // Flush and compaction
+
+    // A 1 or 0 flag indicating whether a memtable flush is pending.
+    // If this number is 1, it means a memtable is waiting for being flushed,
+    // but there might be too many L0 files that prevents it from being flushed.
+    // RocksDB's internal property key: "rocksdb.mem-table-flush-pending"
+    pub mem_table_flush_pending: i64,
+
+    // A 1 or 0 flag indicating whether a compaction job is pending.
+    // If this number is 1, it means some part of the column family requires
+    // compaction in order to maintain shape of LSM tree, but the compaction
+    // is pending because the desired compaction job is either waiting for
+    // other dependnent compactions to be finished or waiting for an available
+    // compaction thread.
+    // RocksDB's internal property key: "rocksdb.compaction-pending"
+    pub compaction_pending: i64,
+
+    // The number of compactions that are currently running for the column family.
+    // RocksDB's internal property key: "rocksdb.num-running-compactions"
+    pub num_running_compactions: i64,
+
+    // The number of flushes that are currently running for the column family.
+    // RocksDB's internal property key: "rocksdb.num-running-flushes"
+    pub num_running_flushes: i64,
+
+    // FIFO Compaction related
+
+    // returns an estimation of the oldest key timestamp in the DB. Only vailable
+    // for FIFO compaction with compaction_options_fifo.allow_compaction = false.
+    // RocksDB's internal property key: "rocksdb.estimate-oldest-key-time"
+    pub estimate_oldest_key_time: i64,
+
+    // Misc
+
+    // The accumulated number of RocksDB background errors.
+    // RocksDB's internal property key: "rocksdb.background-errors"
+    pub background_errors: i64,
+}
+
+impl BlockstoreRocksDbColumnFamilyMetrics {
+    /// Report metrics with the specified metric name and column family tag.
+    /// The metric name and the column family tag is embeded in the parameter
+    /// `metric_name_and_cf_tag` with the following format.
+    ///
+    /// For example, "blockstore_rocksdb_cfs,cf_name=shred_data".
+    pub fn report_metrics(&self, metric_name_and_cf_tag: &'static str) {
+        datapoint_info!(
+            metric_name_and_cf_tag,
+            // Size related
+            (
+                "total_sst_files_size",
```

Comment:
I would test reporting this on a local testnet to see if any of these fields get truncated. I think we've seen datapoints with too many fields lose some of the fields, but I don't recall the exact limit

Reply:
Thanks for bringing this up.  These datapoints are reported for every one second.

Currently, we have around [3k~4k non-zero datapoints](https://metrics.solana.com:8888/sources/1/chronograf/data-explorer?query=SELECT%20COUNT%28%22total_sst_files_size%22%29%20AS%20%22count_total_sst_files_size%22%20FROM%20%22mainnet-beta%22.%22autogen%22.%22blockstore_rocksdb_cfs%22%20WHERE%20time%20%3E%20%3AdashboardTime%3A%20AND%20time%20%3C%20%3AupperDashboardTime%3A%20AND%20%22cf_name%22%3D%27shred_data%27%20AND%20%22total_sst_files_size%22%20%3E%200%20GROUP%20BY%20time%28%3Ainterval%3A%29%20FILL%28null%29) for each time interval, while [very few data points](https://metrics.solana.com:8888/sources/1/chronograf/data-explorer?query=SELECT%20COUNT%28%22total_sst_files_size%22%29%20AS%20%22count_total_sst_files_size%22%20FROM%20%22mainnet-beta%22.%22autogen%22.%22blockstore_rocksdb_cfs%22%20WHERE%20time%20%3E%20%3AdashboardTime%3A%20AND%20time%20%3C%20%3AupperDashboardTime%3A%20AND%20%22cf_name%22%3D%27shred_data%27%20AND%20%22total_sst_files_size%22%20%3D%200%20GROUP%20BY%20time%28%3Ainterval%3A%29%20FILL%28null%29) where we have total_sst_files_size = 0.  Although they might be really those ledger store which just started.

Reply:
I am seeing this errors in the logs:
```
WARN  solana_metrics::metrics] submit response unsuccessful: 413 Payload Too Large <html>
    <head><title>413 Request Entity Too Large</title></head>
    <body>
    <center><h1>413 Request Entity Too Large</h1></center>
    <hr><center>nginx/1.18.0 (Ubuntu)</center>
    </body>
    </html>
```